src/chart/generateCategoricalChart.tsx

```diff
@@ -681,6 +687,7 @@ export const createDefaultState = (props: CategoricalChartProps): CategoricalCha
   const brushItem = findChildByType(children, Brush);
   let startIndex = 0;
   let endIndex = 0;
+
   if (props.data && props.data.length !== 0) {
     endIndex = props.data.length - 1;
```

Comment:
This probably needs to reference the combined graphical items data arrays if they are set. 

We can probably do that in 3.x so this works no matter where data is set. 

@PavelVanecek fyi